cornell-dti/office-hours
# Integrating Text Notifications in Cloud Functions, Updating Design

### Summary 

Migrated the front-end text notification implementation to the cloud functions, and implemented the new design from Figma.








### Test Plan 
I'd recommend following the workflow on Figma, with an extra step for testing the actual text notification sends
- Start from class session dashboard
- Open text notification settings in topbar
- test that both error messages work (can't submit without phone number or consent)
- test that individual error messages work (consent but not phone)
- add phone number and ensure modal closes and banner appears
- add a question to an empty queue and test that you receive a text notif for reaching the top of the queue
- try changing phone numbers (only need to check database for this one) and ensure that "save/disable" design matches
- disable phone number and ensure text notification isn't sent, and that settings design reverts to "enable text notifs"

UPDATED
The only new changes from the banner/snackbar PR is the "don't show again" option on the text notification banner.

### Notes 



### Breaking Changes 

None
